### ISSUE TITLE
Don't set provider features by default

### ISSUE THREAD

**Issue by bluss** [stainless-steel/blas-sys#14]
I'm trying to migrate to blas-sys, it's a bit tricky to navigate the cargo details.

First, I'm gunning for a simple configuration: Netlib, use system libraries.

My libblas provides cblas_* functions, and libcblas does not exist, so I need to turn off default features in the netlib crate, so that include-cblas is turned off. However, this crate's `netlib` feature depends on the default features in `netlib-provider`, so it's not possible.

**Comment by bluss** [comment]
For example, something like this does not succeed right now due to including the include-lapacke, include-cblas features.

```
blas-sys = { version = "*", optional = true, default-features = false, features = ["netlib"] }
netlib-provider = { version = "*", default-features = false, features = ["system-netlib"] }
```

**Comment by bluss** [comment]
It's not super important, since I do have it working nicely with openblas.

**Comment by bluss** [comment]
I'll just keep babbling. 

This configuration could work well for me in ndarray (a library crate)

This dependency is enough to leave the provider user-supplied:

```toml
blas-sys = { version = "0.5", optional = true, default-features = false }
```

Then I can add a more specific feature I can use for testing

```rust
blas = ["blas-sys"]                                                              
blas-openblas-sys = [                                                            
    "blas",                                                                      
    "blas-sys/openblas",                                                         
    "blas-sys/openblas-provider/system-openblas",                                
]
```

**Comment by IvanUkhov** [comment]
Thanks for the issue! I’m very much interested in making this crate ~~a better place~~ suitable for as many use cases as possible, and so I’m totally open to any constructive suggestions.

**Comment by IvanUkhov** [comment]
If you have a justifiably better architecture in mind, we could restructure all the provider, binding, and wrapper crates and their interdependencies as needed.

**Comment by IvanUkhov** [comment]
Related to #5.

**Comment by bluss** [comment]
If you're curious, I'm using blas-sys @ bluss/rust-ndarray/pull/134

I don't have a large architecture to suggest, but I think the blas-sys/netlib feature depending on netlib-provider's default features makes it hard/impossible to opt out of those.

**Comment by IvanUkhov** [comment]
To summarize, currently there are the following solutions:

1. One could reexport features, which is ugly and doesn’t scale that well. Should all subsequent crates do this reexporting then? It’ll snowball.
2. One could disable the default features of the providers, which will probably lead to some confusion among dependants. I’m guessing many want `cblas`, but won’t get it without additional tuning.
3. One could depend on `blas-sys` while disabling its default features and depending directly on one of the providers (or on something else that will do the linking). Haven’t tried, but should work.

**Comment by IvanUkhov** [comment]
cc @cmr

**Comment by IvanUkhov** [comment]
Closed in favor of #17.
